util-flywayinstaller/src/main/java/com/github/cafapi/util/flywayinstaller/Migrator.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.util.flywayinstaller;
+
+import com.github.cafapi.util.flywayinstaller.exceptions.InvalidConnectionStringException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.flywaydb.core.Flyway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Migrator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(Migrator.class);
+    private static final String CONNECTION_URL_REGEX = "^.*\\/\\/.+:\\d+\\/";
+
+    private Migrator()
+    {
+    }
+
+    public static void migrate(
+        final boolean allowDBDeletion,
+        final String connectionString,
+        final String dbName,
+        final String username,
+        final String password
+    ) throws InvalidConnectionStringException, SQLException
+    {
+        LOGGER.debug("Arguments received"
+            + " allowDBDeletion: {}\n"
+            + " connectionString: {}\n"
+            + " dbName: {}\n"
+            + " username: {}\n"
+            + " password: {}", allowDBDeletion, connectionString, dbName, username, password);
+
+        LOGGER.info("Starting migration ...");
+        try (final BasicDataSource dbSource = new BasicDataSource()) {
+            dbSource.setUrl(checkAndConvertConnectionUrl(connectionString));
+            dbSource.setUsername(username);
+            dbSource.setPassword(password);
+
+            final boolean exists = checkDBExists(dbSource, dbName);
+            LOGGER.debug("Database exists: {}", exists);
+            if (!exists || allowDBDeletion) {
+                resetOrCreateDatabase(dbSource, exists, dbName);
+            }
+
+            LOGGER.info("About to perform DB update.");
+            final Flyway flyway = Flyway.configure()
+                .dataSource(dbSource.getUrl() + dbName, username, password)
+                .baselineOnMigrate(true)
```

Comment:
Why are we setting this?

Reply:
This will set the first Flyway migration if not existent. It will initialize the schema history table. https://flywaydb.org/documentation/configuration/parameters/baselineOnMigrate

Reply:
This is to avoid that exception that would occur if the database is not empty: `[12:03:52.104Z #6e8.001 ERROR -            -   ] c.g.c.u.f.Migrator: {"message":"Found non-empty schema(s) \"public\" but no schema history table. Use baseline() or set baselineOnMigrate to true to initialize the schema history table.","exception":"org.flywaydb.core.api.FlywayException: Found non-empty schema(s) \"public\" but no schema history table. Use baseline() or set baselineOnMigrate to true to initialize the schema history table.\r\n\tat org.flywaydb.core.Flyway$1.execute(Flyway.java:154)\r\n\tat org.flywaydb.core.Flyway$1.execute(Flyway.java:124)\r\n\tat org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:214)\r\n\tat org.flywaydb.core.Flyway.migrate(Flyway.java:124)\r\n\tat com.github.cafapi.util.flywayinstaller.Migrator.migrate(Migrator.java:75)\r\n\tat com.github.cafapi.util.flywayinstaller.Application.call(Application.java:96)\r\n\tat com.github.cafapi.util.flywayinstaller.Application.call(Application.java:27)\r\n\tat picocli.CommandLine.executeUserObject(CommandLine.java:1953)\r\n\tat picocli.CommandLine.access$1300(CommandLine.java:145)\r\n\tat picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)\r\n\tat picocli.CommandLine$RunLast.handle(CommandLine.java:2352)\r\n\tat picocli.CommandLine$RunLast.handle(CommandLine.java:2314)\r\n\tat picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)\r\n\tat picocli.CommandLine$RunLast.execute(CommandLine.java:2316)\r\n\tat picocli.CommandLine.execute(CommandLine.java:2078)\r\n\tat com.github.cafapi.util.flywayinstaller.Application.main(Application.java:84)\r\n"}`

We don't have any baselineVersion to refer to at the moment as it will be our first migration. We could potentially add another argument later on to specify it.